### ISSUE TITLE
[Bug] Reopen - Data loss with crash or incorrect game saving

### ISSUE THREAD

**Issue by asandyguy** [Ryujinx/Ryujinx#4516]
### Description of the issue

oldversion case is closed, sorry to reopen this issue, goodsync is garbage for insurance.

There is a problem with saving in a few games, so far only Zelda Breath of the Wild and Star Kirby, sav. can be read immediately, but the data will be lost after exiting game. Not sure if autosave has this problem.

my god can anyone help me out?  really need a safety insurance against game crash with data loss!

### Reproduction steps

1. Select a manual file to read - start game
2. Save the game
3. Ensure Loading and saving interface displayed normally
4. Exit the game or close Ryujinx completely, next play, the sav. will lost
5. These pics specifically show how to locate the file,
and the process difference between two data after the game is saved. One of the pictures shows that the files in the \0 directory are not overwritten correctly, it will be restored after crashing or exiting the emu process.

![detail1](https://user-images.githubusercontent.com/124961320/218004287-3d3154fa-a51c-4a78-99d6-8450fb1a7715.png)
![detail2](https://user-images.githubusercontent.com/124961320/218004317-c93fcbbc-1e68-416b-bd76-b4c776994aed.png)
![detail3](https://user-images.githubusercontent.com/124961320/218004361-5d42d5a5-30d0-4da3-bcec-a9a9b9f29406.png)
![detail4](https://user-images.githubusercontent.com/124961320/218004415-e973cffe-a66c-4f05-b269-955e54ae5cfd.png)

### Log file

[Ryujinx_1.1.610_2023-02-10_log1.log](https://github.com/Ryujinx/Ryujinx/files/10703804/Ryujinx_1.1.610_2023-02-10_log1.log)

### OS

Windows 11 home 21h2 beta

### Ryujinx version

1.1.610 or higher

### Game version

zelda botw 1.6.0

### CPU

12th Gen Intel (R) i5 -12500h 2.50Ghz 45w

### GPU

Nvidia RTX2050 4GB 45w

### RAM

pm9a1 16GB

### List of applied mods

no mods

### Additional context?

[Zelda BOTW save test.zip](https://github.com/Ryujinx/Ryujinx/files/10711424/Zelda.BOTW.save.test.zip)

**Comment by asandyguy** [comment]
need help please

**Comment by asandyguy** [comment]
i will test on 645 later, and upload video about this

**Comment by Squall-Leonhart** [comment]
Your save games are in a location that is backed up (or in this case, reverted) by a cloud file provider.

**Comment by asandyguy** [comment]
No, I use goodsync just for backup. It will copy saved files to the backup directory. After closing the emulator process, the original data will be lost.

**Comment by Thealexbarney** [comment]
To be clear, is this what happens?
1. Start the game and load your existing save
2. Save the game
3. Close the game
4. Start the game again and load the save
5. The loaded save is the save that existed prior to step 1, not the save that was made in step 2

If this is what happens, does the same thing happen when you disable and close goodsync?

**Comment by asandyguy** [comment]
As you said, the thing happens like this. 

It happened without goodsync and without mod.

I haven't run ryujinx one month for other reasons,I suspect this bug may happen after ryujinx crashed, strangely this problem seems to be reproduced even if you delete game data and restart the game.

**Comment by FireN1337** [comment]
Sadly, this happened to me witch Zelda TOTK today, I played the game for 23h23m worth of savegame :(
My pc crashed ingame and had to restart.
After restarting I wanted to play on and the game started like I never played it, but the time played was shown correctly.
I instantly checked the save files and all files (in file 0 and 1) are overwritten at the same time and my saves are lost...

**Comment by GrantIsEaton** [comment]
Just wanted to share that this happened to me, as well, today with TOTK. PC crashed due to unknown reasons (unsure if caused by my gpc or ryu) and when i rebooted the game started up as if i never played it. Checked the game saves and everything was gone. No mods. 
I'm really curious to learn why this might have happened. For this to have happened, I think that ryu would have had to decided to clear out 100% of the save files for some reason. I could understand, for example, if a crash would corrupt the most recent save file for some reason if it cut out while saving the game, but why would it erase prior saves as well?

Anyways, I was 7 shrines from all 152 and at 300+ koroks, guess that's where this will end 👍

**Comment by julsb00** [comment]
This happened to me today with Zelda TOTK. I played the game for 10h.
PC crashed due to unknown reasons. After restarting the game started like I never played it. I checked the save files and files in file 0 and 1 are overwritten at the same time and my saves are lost... No mods or anything.

**Comment by artur-shaik** [comment]
Same for `Mario Bros. Wonder` and `Kirby Star Allies`

**Comment by caiohenrqq** [comment]
same for totk, just lost everything AND i have the backup, but the backup dont work for some reason.
